File: GFG/Count of Subarrays/Count of Subarrays.java
class Solution {

    long countSubarray(int arr[], int n, int k) {
        long ans=0;
        int recent=-1;
        for(int i=0;i<n;i++){
            if(arr[i]>k){
                ans++;
                int left=i-(recent+1);
                int right=n-1-i;
                ans+=left;
                ans+=right;
                ans+=(left*right);
                recent=i;
            }
        }
        return ans;
    }
}

File: GFG/Count of Subarrays/Count of Subarrays.cpp
class Solution{
public:
	// #define ll long long

	ll countSubarray(int arr[], int n, int k) {
	       // code here

     ll ans = (ll)n*(n+1)/2;

        ll l =0;

        for(int i=0;i<n;i++) {

            if(arr[i]<=k) {

                l++;

            } else {

                ans-=(l*(l+1)/2);

                l=0;

            }

        }

        ans-=(l*(l+1)/2);

        return ans;
	}
};
